src/value/generator.rs

```diff
@@ -5,19 +5,26 @@
 // https://github.com/maciejhirsz/json-rust/blob/master/src/codegen.rs
 
 use crate::value::ValueTrait;
-#[cfg(target_arch = "x86")]
-use std::arch::x86::*;
-#[cfg(target_arch = "x86_64")]
-use std::arch::x86_64::*;
 use std::io;
 use std::io::Write;
 use std::marker::PhantomData;
 use std::ptr;
 
+#[cfg(target_arch = "x86")]
+use std::arch::x86::*;
+#[cfg(any(target_feature = "sse4.2", target_feature = "avx2"))]
```

Comment:
this should be `#[cfg(target_arch = "x86_64")]` otherwise it will be double imported on a 32bit OS w/ avs2 or see4.2